### DISCUSSION TITLE
Cannot use BaseModel.model_dump(exclude_unset=True) when getting models from query params

### DISCUSSION THREAD

**Discussion by Dest0re** [fastapi/fastapi#10217] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated
from fastapi import Depends, FastAPI
from pydantic import BaseModel

app = FastAPI()


class Item(BaseModel):
    name: str | None = None
    description: str | None = None


@app.get("/items/")
async def get_items(filter: Annotated[Item, Depends()]):
    return filter.model_dump(exclude_unset=True)
```

### Description

* Open the browser and call the enpoint '/items/' with NO query params
* Returns following:
```json
{
    "name": null,
    "description": null
}
```
* Although I expected it to return an empty json because exclude_unset were submitted.

### Operating System

Linux

### FastAPI Version

0.101.0

### Pydantic Version

2.1.1

### Python Version

3.11.3

**Comment by Housain-maina** [comment]
That is because both name and description fields in your pydantic model have a default value of None. If you wish to return an empty json, you'll have to remove the default value of None you set for both fields.

**Comment by YuriiMotov** [comment]
Duplicate: https://github.com/fastapi/fastapi/discussions/9595
